src/Polly.Core/Hedging/Controller/TaskExecution.cs

```diff
@@ -137,6 +139,21 @@ public void Cancel()
         return true;
     }
 
+    private async Task HandleOnHedgingAsync(ResilienceContext primaryContext, int attemptNumber)
+    {
+        var args = new OnHedgingArguments<T>(
+            primaryContext,
+            Context,
+            attemptNumber);
+
+        _telemetry.Report(new(ResilienceEventSeverity.Warning, HedgingConstants.OnHedgingEventName), Context, args);
+
+        if (_handler.OnHedging is not null)
```

Comment:
What about capturing it into a local like `is {} callback` and then invoking that?